### PR TITLE
Scaffold the initial app

### PR DESCRIPTION
**Pull request by sbh69840**
This PR will clean up the repo to have only the haskell project. Further PRs will build on the base created by this PR to nixify the project.

### CODE REVIEW

File: src/Request.hs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE DataKinds #-}
+
+module Request (done, view, add, delete, reset) where
```
Comment (srid) [inline src/Request.hs:6]: `module TodoApp.Request`

File: src/Request.hs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      HttpMethod, jsonResponse, responseBody, HttpBodyAllowed, AllowsBody, CanHaveBody (CanHaveBody))
+import Text.URI (mkURI)
+import Data.Maybe (fromJust)
+import Data.Text (concat, pack)
+import Prelude hiding (concat)
+import Data.Aeson (Value, ToJSON)
+import Data.Kind (Type)
+import Control.Monad.IO.Class (MonadIO)
+import GHC.IO (liftIO)
+import Relude (lookupEnv, toText, fromMaybe, readMaybe)
```
Comment (srid) [inline src/Request.hs:26]: I thought you were not going to use relude. But if you are going to use relude, I suggest configuring it as a mixin, so it need not be explicitly imported.

File: src/Request.hs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+data Response = Response {
+   message :: Value
+  ,responseCode :: Int
```
Comment (srid) [inline src/Request.hs:30]: I suppose we would run ormolu in latter PRs.

**Review by srid — COMMENTED** [review]
> This PR will clean up the repo to have only the haskell project.

All of Nix & CI will be removed this PR, IIUC? 

How will you test it? By running postgres/postgrest manually by side?

File: src/Request.hs (hunk 2 above)
Comment (sbh69840) [reply src/Request.hs:26]: I am going to remove relude dependency as well, needed it for lookupEnv mainly. Once I move the lookupEnv to main I can use the one exported by System.Environment that returns an `IO (Maybe String)`

File: src/Request.hs (hunk 3 above)
Comment (sbh69840) [reply src/Request.hs:30]: Yes, I will be running them.

**Comment by sbh69840** [comment]
> How will you test it? By running postgres/postgrest manually by side?
Yes, that's the plan. Want to start building on top of it from scratch!

File: src/Request.hs (hunk 2 above)
Comment (sbh69840) [reply src/Request.hs:26]: I should add a TODO there

File: src/TodoApp/Request.hs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  (HttpBodyAllowed (AllowsBody method) 'CanHaveBody,
+   MonadIO m,
+   HttpMethod method,
+   ToJSON a) =>
+   a -> method -> m Response
+request payload method = 
+  runReq defaultHttpConfig $ do
+    -- TODO: Move domain and port to main and use IORef to Store the state 
+    domain <- fromMaybe "http://localhost" <$> lookupEnv "TODO_DOMAIN"
+    todoPort <- ( fromMaybe 3000 <$> readMaybe ) . fromMaybe "3000" <$> lookupEnv "TODO_PORT"
```
Comment (abshekh) [inline src/TodoApp/Request.hs:42]: we can reduce this to
```
todoPort <- fromMaybe 3000 <$> (readMaybe =<<) <$> lookupEnv "TODO_PORT"
```

File: .gitignore [hunk 5]
```diff
@@ -9,4 +9,12 @@ result
...
\ No newline at end of file
+.direnv
+
+# postgres
+.nix
+.s.PGSQL.5432.lock
+logfile
+
+# postgrest
+tutorial.conf
```
Comment (srid) [inline .gitignore:20]: Why not have these in `./db` or something and ignore that dir?

**Review by srid — APPROVED** [review]

File: .gitignore [hunk 6]
```diff
@@ -4,9 +4,5 @@ dist-*
...
 
-# nix
-result
-result-*
-
-# direnv
-.direnv
\ No newline at end of file
+# postgrest
+tutorial.conf
```
Comment (srid) [inline .gitignore:8]: `data/db.conf` and then we just gitignore `./data`

File: .gitattributes [hunk 7]
```diff
@@ -1 +1,7 @@
 flake.lock linguist-generated=true
+# To maintain consistency of line endings across OSs
+*.md       eol=lf
+*.hs       eol=lf
+*.yaml     eol=lf
+*.json     eol=lf
+*.cabal    eol=lf
```
Comment (srid) [inline .gitattributes:7]: remove

File: .vscode/extensions.json [hunk 8]
```diff
@@ -6,6 +6,5 @@
     "arrterian.nix-env-selector",
     "bbenoist.nix",
     "jnoortheen.nix-ide",
-    "tamasfe.even-better-toml"
   ]
-}
```
Comment (srid) [inline .vscode/extensions.json:11]: remove `.vscode` except for those things we need

File: README.md [hunk 9]
```diff
@@ -1,33 +1,21 @@
-# haskell-template
+# todo-app
 
-Get a Haskell development environment up and running quickly. Thanks to Nix, this template is optimized for a fully reproducible and friendly development environment. It is based on:
+Refer to [todo-app.cabal](todo-app.cabal) for description.
```
Comment (srid) [inline README.md:3]: > WIP: A simple todo app in Haskell demonstrating ____. Ultimate goal is ___.

**Pull request merged**